zerver/tests/test_home.py

```diff
@@ -259,7 +259,7 @@ def test_home(self) -> None:
         self.client_post("/json/bots", bot_info)
 
         # Verify succeeds once logged-in
-        with self.assert_database_query_count(53):
+        with self.assert_database_query_count(55):
```

Comment:
The query count increase of 2 is - one due to refetching realm and one for `can_create_public_streams` as it uses the `realm` object from `user_profile` object, we can update that function to just accept `realm` as argument.

Reply:
Yeah passing that is definitely worth doing as a follow-up.